api/workflow/v1alpha1/types.go

```diff
@@ -274,6 +289,10 @@ type NodeStatus struct {
 	// Daemoned tracks whether or not this node was daemoned and need to be terminated
 	Daemoned *bool `json:"daemoned,omitempty"`
 
+	// MaxRetries indicates how many retries should be executed before marking the node
+	// as Failed. MaxRetries is the same as BackoffLimit from the template.
+	MaxRetries int32 `json:"maxRetries,omitempty"`
```

Comment:
I don't understand. If the user wants 'run-to-completion' semantics with this feature, either the backoffLimit will have to be set to some really high number (which still won't guarantee successful completion, but increase the probability of at least one run completing successfully) or special logic will have to be added for infinite retries (say, when the backoffLimit is set to -1).

Let me know if I'm missing something.

Reply:
> special logic will have to be added for infinite retries (say, when the backoffLimit is set to -1)

Yes you're right, we would need special logic since backoffLimit is expected to be > 0. I was also thinking of the magic -1 value for infinite retries.

However, after looking more closely of the term backoffLimit with respect to Jobs, I now think we need to use different terminology. Jobs by their very nature are run-to-completion with automatic retries, no ability to opt-out -- jobs are expected to execute containers who are eventually expected to complete with exit code 0. backoffLimit does not equal retries because in jobs, backoffLimit reset "if no new failed Pods appear before the Job’s next status check." We would have no such concept. We should not use backoffLimit to avoid confusion with jobs since they are inherently different concepts. Retry is more appropriate.